tests/common/mod.rs

```diff
@@ -0,0 +1,384 @@
+//! Utility code to help writing triagebot tests.
+
+use std::collections::HashMap;
+use std::io::{BufRead, BufReader, Read, Write};
+use std::net::TcpStream;
+use std::net::{SocketAddr, TcpListener};
+use std::sync::{Arc, Mutex};
+use url::Url;
+
+/// The callback type for HTTP route handlers.
+pub type RequestCallback = Box<dyn Send + Fn(Request) -> Response>;
+
+/// HTTP method.
+#[derive(Copy, Clone, Debug, PartialEq, Eq, Hash)]
+pub enum Method {
+    GET,
+    POST,
+    PUT,
+    DELETE,
+    PATCH,
+}
+
+impl Method {
+    fn from_str(s: &str) -> Method {
+        match s {
+            "GET" => Method::GET,
+            "POST" => Method::POST,
+            "PUT" => Method::PUT,
+            "DELETE" => Method::DELETE,
+            "PATCH" => Method::PATCH,
+            _ => panic!("unexpected HTTP method {s}"),
```

Comment:
The reason I didn't use it is because:
* I wanted an enum to make it easy to `use Method::*`. `http` uses associated consts.
* I also wanted it to impl `Copy` to make it easy to use.
* I had a custom STOP verb, which is not standard, and the `http` Method extensions weren't easy to use. I could easily switch that to use a different method, though.

I understand that it might be nicer to not use a custom http server, but I suspect adding anything else would have a pretty large build time and complexity hit. 

I could trim a few things down, like removing custom header support. It isn't being used here, and I'm not sure it will ever be needed (I copied this from Cargo which does use it). However, it only adds a few lines so I figured it didn't hurt to keep it.